### ISSUE TITLE
Should ParseArrayPipe with String items treat it as JSON?

### ISSUE THREAD

**Issue by thgh** [nestjs/nest#11173]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

`new ParseArrayPipe({ items: String })` interprets each item as JSON and then stringifies it which potentially modifies it

### Minimum reproduction code

See below or #11170

### Steps to reproduce

```ts
Query('example', new ParseArrayPipe({ items: String })) example: string[]
```

```ts
URL: ?example=1,2.0,3

example = ["1", "2", "3"]
```

### Expected behavior

```ts
example = ["1", "2.0", "3"]
```

### Package

- [ ] I don't know. Or some 3rd-party package
- [X] <code>@nestjs/common</code>
- [ ] <code>@nestjs/core</code>
- [ ] <code>@nestjs/microservices</code>
- [ ] <code>@nestjs/platform-express</code>
- [ ] <code>@nestjs/platform-fastify</code>
- [ ] <code>@nestjs/platform-socket.io</code>
- [ ] <code>@nestjs/platform-ws</code>
- [ ] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [ ] Other (see below)

### NestJS version

9

### Packages versions

```json
core version             : 9.3.9
```

### Node.js version

16

### In which operating systems have you tested?

- [X] macOS
- [ ] Windows
- [X] Linux

### Other

The behaviour is untested, so I added the case in #11170

**Comment by kamilmysliwiec** [comment]
Would you like to create a PR for this issue?

**Comment by depak379mandal** [comment]
@thgh You are right below I shared the link of the code that is actually doing this. So you will be not wasting time on finding. Have a nice day.


https://github.com/nestjs/nest/blob/master/packages/common/pipes/parse-array.pipe.ts#L92

**Comment by thgh** [comment]
Haha, [same thinking here](https://github.com/nestjs/nest/pull/11170#issuecomment-1446313103)

**Comment by thgh** [comment]
> Would you like to create a PR for this issue?

Here is the PR: #11170

I'm not sure what the expected outcome is though. Can I remove the JSON.parse? Or fiddle until the tests pass?

**Comment by micalevisk** [comment]
@thgh 

> I'm not sure what the expected outcome is though

given `?example=1,2.0,3` and `new ParseArrayPipe({ items: String })`, we should get `["1", "2.0", "3"]` instead of `["1", "2", "3"]`

Would you like to open a PR to address this?

**Comment by CodyTseng** [comment]
I think we can remove `JSON.parse` only when `items` is `String`.

**Comment by thgh** [comment]
Alright, the new behaviour has been pushed to https://github.com/nestjs/nest/pull/11170

**Comment by kamilmysliwiec** [comment]
Let's track this here https://github.com/nestjs/nest/pull/11170
